Backend/nix/services.nix

```
@@ -0,0 +1,122 @@
+{ inputs, self', ... }:
+{
+  config = {
+    perSystem = { inputs', self', pkgs, lib, ... }: {
+      process-compose."services" = { config, ... }:
+        let
+          startScript = masterDb: replicaDb: port: pkgs.writeShellApplication {
+            name = "start-pgbasebackup";
+            runtimeInputs = with pkgs; [ config.services.postgres."${masterDb}".package coreutils ];
+            text = ''
+              masterDbPath=$(readlink -f data/${masterDb})
+              replicaDbPath=$(readlink -f data/${replicaDb})
+              pg_basebackup -h "$masterDbPath" -U repl_user --checkpoint=fast -D "$replicaDbPath"  -R --slot=some_name  -C --port=${port} 
```

port should be integer. So,

```suggestion
              pg_basebackup -h "$masterDbPath" -U repl_user --checkpoint=fast -D "$replicaDbPath"  -R --slot=some_name  -C --port=${builtins.toString port} 
```